iothub/service/design.md

```diff
@@ -0,0 +1,119 @@
+# RBAC API Design for IoT hub
+
+## Service Client
+
+```csharp
+/// <summary>
+/// Create ServiceClient using AAD token credentials and the specified Transport Type.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential">AAD <see cref="TokenCredential"> to authenticate with IoT hub.</param>
+/// <param name="transportType">Specifies whether Amqp or Amqp_WebSocket_Only transport is used.</param>
+/// <param name="transportSettings">Specifies the AMQP and HTTP proxy settings for Service Client.</param>
+/// <param name="options">The options that allow configuration of the service client instance during initialization.</param>
+/// <returns>An instance of <see cref="ServiceClient"/>.</returns>
+public static ServiceClient Create(
+    string hostName,
+    TokenCredential credential,
+    TransportType transportType,
+    ServiceClientTransportSettings transportSettings = default,
+    ServiceClientOptions options = default)
+
+/// <summary>
+/// Create ServiceClient using SAS token credentials and the specified Transport Type.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential"><see cref="AzureSasCredential"> to authenticate with IoT hub.</param>
+/// <param name="transportType">Specifies whether Amqp or Amqp_WebSocket_Only transport is used.</param>
+/// <param name="transportSettings">Specifies the AMQP and HTTP proxy settings for Service Client.</param>
+/// <param name="options">The options that allow configuration of the service client instance during initialization.</param>
+/// <returns>An instance of <see cref="ServiceClient"/>.</returns>
+public static ServiceClient Create(
+    string hostName,
+    AzureSasCredential credential,
+    TransportType transportType,
+    ServiceClientTransportSettings transportSettings = default,
+    ServiceClientOptions options = default)
+```
+
+## Registry Manager
+
+```csharp
+/// <summary>
+/// Creates an instance of <see cref="RegistryManager"/>.
+/// </summary>
+/// <param name="hostName">IoT hub host name.</param>
+/// <param name="credential">AAD <see cref="TokenCredential"> to authenticate with IoT hub.</param>
+/// <param name="transportSettings">The HTTP transport settings.</param>
+/// <returns>An instance of <see cref="RegistryManager"/>.</returns>
+public static RegistryManager Create(
+    string hostName,
+    TokenCredential credential,
```

Comment:
Maybe nitpicky, but to me this argument is more of a "credentialProvider" or "authenticationTokenProvider" rather than a "credential". What do you think though?

Reply:
If that's what it is, why is `TokenCredential` named as it is?

In our track2 ADT code we call this `credential` too. Why not stick with that standard?

Reply:
> If that's what it is, why is TokenCredential named as it is?

No clue. I can't say that I like it's name, to be honest.

> In our track2 ADT code we call this credential too. Why not stick with that standard?

I can get on board with this point. Since we plan on using implementations from the Azure Identity library anyways, we may as well follow their naming patterns